.github/workflows/pr-preview-build.yml

```diff
@@ -0,0 +1,50 @@
+# This workflow will run on each PR - it will build the project
+# and upload as artifacts for the deploy workflow
+name: Build PR preview
+
+on:
+  # Runs on pull requests targeting the default branch
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+concurrency:
+  group: preview-${{ github.ref }}
+  cancel-in-progress: false
+
+jobs:
+  build-preview:
+    runs-on: ubuntu-latest
+    steps:
+      # Checkout the branch
+      - name: Checkout
+        uses: actions/checkout@v3
+      
+      # Setup Poetry for dependencies
+      - name: Setup Poetry
+        uses: Gr1N/setup-poetry@v8
+      
+      # Build the site with Poetry
+      - name: Build html
+        run: |
+          cd docs
+          poetry install
+          poetry run make html
+      
+      # Store PR number so that it can be extracted in Deploy
+      - name: Create PR artifact
+        run: |
+          mkdir -p ./pr
+          echo ${{ github.event.number }} > ./pr/NR
+          mv docs/_build/html ./pr/
```

Comment:
Is the really the best way to pass simple values from one job to another in Github Actions? Gitlab CI has environment passing, maybe Github Actions has something similar?

Reply:
Agreed, I would like a different solution using e.g. env vars more as well - since the file could potentially be used for some kind of code/information injection.
I couldn't find anything like this, especially since both workflows run on different branches - so maybe this isn't possible in GitHub Actions. The documentation even suggests to use text files in artifacts in this way: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#using-data-from-the-triggering-workflow

Reply:
Alright then just move on and merge, thank you a lot :)